inference-engine/src/vpu/graph_transformer/src/stages/roi_align.cpp

```diff
@@ -77,47 +88,57 @@ class ROIAlignStage final : public StageNode {
         }
 
         outputEdge(0)->output()->serializeBuffer(serializer);
-        const auto use_buffer = !tempBuffers().empty();
-        if (use_buffer)
-            tempBuffer(0)->serializeBuffer(serializer);
     }
 };
 
 }  // namespace
 
 void FrontEnd::parseROIAlign(const Model& model, const ie::CNNLayerPtr& layer, const DataVector& inputs, const DataVector& outputs) const {
-    VPU_THROW_UNLESS(inputs.size() == 3,
-                    "ROIAlign stage with name {} has invalid number of inputs: expected 3, "
+    VPU_THROW_UNLESS(inputs.size() == 3 || inputs.size() == 1,
+                    "ROIAlign stage with name {} has invalid number of inputs: expected 3 or 1 "
                     "actually provided {}", layer->name, inputs.size());
-
     VPU_THROW_UNLESS(outputs.size() == 1,
                     "ROIAlign stage with name {} has invalid number of outputs: expected 1, "
                     "actually provided {}", layer->name, outputs.size());
 
-    const auto stage = model->addNewStage<ROIAlignStage>(layer->name, StageType::ROIAlign, layer, inputs, outputs);
     const auto mode = layer->GetParamAsString("mode", "");
+    const auto width = inputs[0]->desc().dim(Dim::W);
+    const auto is_input_static = (inputs[0]->parentDataToShapeEdge() == nullptr);
+    const auto use_chwc_repacking = (width >= 200) && (mode == "avg") && is_input_static;
+    auto repackedInput = inputs[0];
 
+    ROIAlignMode roi_align_mode;
     if (mode == "avg") {
-        stage->attrs().set<ROIAlignMode>(s_mode, ROIAlignMode::Average);
+        roi_align_mode = ROIAlignMode::Average;
     } else if (mode == "max") {
-        stage->attrs().set<ROIAlignMode>(s_mode, ROIAlignMode::Max);
+        roi_align_mode = ROIAlignMode::Max;
     } else {
         VPU_THROW_FORMAT("Layer with name {} supports only (avg, max) mode", layer->name);
     }
```

Comment:
Nitpick: I'd prefer more laconic version:

```cpp
const auto mode = layer->GetParamAsString("mode");
VPU_THROW_UNLESS(mode == "avg" || mode == "max", error_message);
const auto roi_align_mode = mode == "avg" ? ROIAlignMode::Average : ROIAlignMode::Max;

use roi_align_mode instead of mode since it's typed
```